pilot/cmd/pilot-agent/config.go

```diff
@@ -59,7 +63,19 @@ func getTLSCerts(pc meshconfig.ProxyConfig) []string {
 }
 
 func constructProxyConfig() (meshconfig.ProxyConfig, error) {
-	meshConfig, err := getMeshConfig()
+	annotations, err := readPodAnnotations()
+	if err != nil {
+		log.Warnf("failed to read pod annotations: %v", err)
+	}
+	var fileMeshContents string
+	if fileExists(meshConfigFile) {
+		contents, err := ioutil.ReadFile(meshConfigFile)
+		if err != nil {
+			return meshconfig.ProxyConfig{}, fmt.Errorf("failed to read mesh config file %v: %v", meshConfigFile, err)
+		}
+		fileMeshContents = string(contents)
+	}
+	meshConfig, err := getMeshConfig(fileMeshContents, annotations[inject.MeshConfigAnnotation])
```

Comment:
Where is this `meshConfig` used? 

Also I noticed there are some missing fields in ProxyConfig compared to MeshConfig - For example file access log https://github.com/istio/api/blob/master/mesh/v1alpha1/config.proto#L151. What do we do with those fields?

Reply:
It's passed to the agent to configure things like access logs service, tracing, etc. This pr is to allow per proxy overrides of these

I am not sure what you mean about missing fields. ProxyConfig is a field in mesh config. Currently the proxy only uses the proxy config field, but I have chosen to pass the full mesh config in case we need to change that in the future - see linked issue for some additional thoughts

Reply:
> but I have chosen to pass the full mesh config in case we need to change that in the future 

Actually I am also looking for that - but probably missing some thing here. That is why this question. Where you passing the full mesh config to agent? Can you please point me to that?

My question specifically was
- We have `access_log_file` field in MeshConfig. Let us say user gives a value to this field in the MeshConfig
- We are getting ProxyConfig out of this `MeshConfig` in Line 84 and filling other fields and passing to Agent.

Since `access_log_file` is not available in `ProxyConfig` - how will we populate?

BTW, I agree passing full mesh config is the right thing to do for future.

Reply:
I understand now. Basically right now the agent never uses any fields outside of proxy config, so in line 84 we throw away all of the other stuff. So if we wanted to use the access log file somewhere, we would just need to change this to return the full thing.

We are passing the config to the agent by injection as an env var for sidecar and mounting for gateway

Reply:
> we would just need to change this to return the full thing.

Yeah. I agree. We can probably do once this lands?

Reply:
Its a trivial change - no need to make it until we want to use it. At some point in the code we need to ignore the mesh config and just use the proxy config so it doesn't really matter where that is right now in my opinion

Reply:
OK. I see the difference now. When I read the description specifically this `The goal here is to allow overrides of mesh config per workload` - I was assuming that we read the mesh config from file/what ever and pass it to Pilot via Node Metadata and pilot uses it apply workload specific settings. It seems that is not the case - it is to just allow/override ProxyConfig part from MeshConfig. 

Is that correct?